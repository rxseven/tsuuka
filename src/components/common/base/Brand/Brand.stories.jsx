import React from 'react';
import { storiesOf } from '@storybook/react';

import Brand from './index';

const title = 'Components/Common/Base/Brand';
const component = <Brand />;

storiesOf(title, module).add('default', () => component);
