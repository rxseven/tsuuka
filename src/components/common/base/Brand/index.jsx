import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Logo = styled(Link)`
  display: block;
  font-size: 1.1rem;
  margin-bottom: 0;
  padding: 0;
  transition: color 0.2s ease-in-out;

  &,
  :hover {
    color: ${({ theme }) => theme.color.dark};
  }

  :hover {
    color: ${({ theme }) => theme.color.primary};
    text-decoration: none;
  }
`;

function Brand() {
  return (
    <Logo title="Currency Converter" to="/">
      通貨換算ツール
    </Logo>
  );
}

export default Brand;
