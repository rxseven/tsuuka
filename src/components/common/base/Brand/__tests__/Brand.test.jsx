import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Environment from 'tests/environment';
import Brand from '../index';

describe('<Brand />', () => {
  // Arrange
  const props = { children: 'Link Text', href: '/' };
  const component = (
    <Environment>
      <Brand {...props} />
    </Environment>
  );

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
