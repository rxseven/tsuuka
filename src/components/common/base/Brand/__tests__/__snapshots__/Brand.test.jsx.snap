// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brand /> Snapshot tests should render correctly 1`] = `
<a
  className="sc-bdVaJa gRJttW"
  href="/"
  onClick={[Function]}
  title="Currency Converter"
>
  通貨換算ツール
</a>
`;
